React/Testing best practices.md

```diff
@@ -585,3 +624,41 @@ describe("useModal", () => {
 ```
 
 If any part of your test is preforming an update, that action needs to be wrapped into `act()`. More info about [act()](https://reactjs.org/docs/test-utils.html#act).
+
+### Mock hooks
+
+Sometimes we want our custom hook to return a mock response while we test component consuming it.
+
+Example hook:
+
+```tsx
+// @/hooks/useTodos.ts
+export const useTodos = (config?: SWRConfiguration) => {
+  return useSWR(() => {
+    return "api/todo-lists";
+  }, config);
+};
+```
+
+Example mock and test:
+
+```tsx
+import * as hooks from "@/hooks/useTodos";
+
+const mockTodoData = [{ title: "Todo #1", todos: [] }];
+
+jest.spyOn(hooks, "useTodos").mockImplementation(() => {
```

Comment:
This way I've avoided struggling with path aliases in `next/jest` while using `jest.mock`